### PR TITLE
Rollout Jinja2 template improvements to CSLC-S1 PGE

### PR DESCRIPTION
**Pull request by RKuttruff**
## Description
- Rolled out templating improvements done in #496 to CSLC-S1 PGE

## Affected Issues
- Completes #523 
  - Part of #520

## Testing
- Manually diffed ISO XML for both standard and static configs from unit test runs between this and main branch
- Ran INT test for PGE on opera-dev-pge and reviewed the XML outputs

### CODE REVIEW

File: src/opera/pge/cslc_s1/cslc_s1_pge.py [hunk 1]
```diff
@@ -601,6 +603,33 @@ def _qa_log_filename(self):
...
+
+            while len(key_path) > 0:
+                mp = mp[key_path.pop(0)]
+
+            new_measured_parameters[parameter_var_name] = mp
+
+        augmented_parameters = super().augment_measured_parameters(new_measured_parameters)
+
+        return augmented_parameters
+
```
Comment (RKuttruff) [inline src/opera/pge/cslc_s1/cslc_s1_pge.py:632]: Should probably move this to a util module.

Something like flatten_h5_metadata_by_config

File: src/opera/pge/cslc_s1/cslc_s1_pge.py [hunk 2]
```diff
@@ -622,6 +651,10 @@ def _collect_cslc_product_metadata(self, cslc_product):
         # Extract all metadata assigned by the SAS at product creation time
         try:
             output_product_metadata = get_cslc_s1_product_metadata(cslc_product)
+            output_product_metadata['MeasuredParameters'] = self.augment_measured_parameters(
+                # flatten_h5_metadata(output_product_metadata)
```
Comment (collinss-jpl) [inline src/opera/pge/cslc_s1/cslc_s1_pge.py:655]: Is this comment a left-over placeholder? If so we can just remove it.

**Review by collinss-jpl — APPROVED** [review]
Only had some minor findings. Feel free to merge this branch once they've been addressed. Thanks @RKuttruff !

File: src/opera/pge/cslc_s1/cslc_s1_pge.py (hunk 2 above)
Comment (RKuttruff) [reply src/opera/pge/cslc_s1/cslc_s1_pge.py:655]: It absolutely is a leftover from my previous approach at this

**Pull request merged**